README.rst

```diff
@@ -391,6 +391,58 @@ In case you just want to change the colors used in a chart here's how to do it:
         }
     }
 
+``OPENWISP_MONITORING_METRICS``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++--------------+-------------+
+| **type**:    | ``dict``    |
++--------------+-------------+
+| **default**: | ``{}``      |
++--------------+-------------+
+
+This setting allows to define additional metric configuration or to override
+the default metric configuration defined in
+``openwisp_monitoring.monitoring.metrics.DEFAULT_METRICS``.
+
+For example, if you want to change the field_name of
+``clients`` metric to ``wifi_clients`` you can use:
+
+.. code-block:: python
+
+    from django.utils.translation import ugettext_lazy as _
+
+    OPENWISP_MONITORING_METRICS = {
+        'clients': {
+            'label': _('Clients'),
+            'key': '{key}',
+            'field_name': 'wifi_clients',
+        },
+    }
+
+**NOTE**: If you want to use the same metric configuration for multiple metrics, it can be done as follows:
+
+In the custom configuration in ``OPENWISP_MONITORING_METRICS``, ``key`` and ``field_name`` should be
+**{key}** and **{field_name}** respectively.
+By doing so, these values shall be automatically taken from the ``Metric`` name.
+
+For example, you want to add a new metric configuration named **resources** and use it for multiple metrics,  it can be done as below:
+
+.. code-block:: python
+
+    from django.utils.translation import ugettext_lazy as _
+
+    OPENWISP_MONITORING_METRICS = {
+        'resources': {
+            'label': _('Resources'),
+            'key': '{key}',
+            'field_name': '{field_name}',
+        },
+    }
+
+If the metric with ``resources`` configuration is named as **Resources Memory** then its ``key``
+would be **Resources** and ``field_name`` shall be **Memory** respectively.
+If you name it as **Memory**, then both ``key`` and ``field_name`` shall have the same value **Memory**
+
```

Comment:
I find it simpler and clearer if the user defines their own key and field_name values, instead of guessing them from the metric name. Did you take this approach for any special reason?

Reply:
That option is always open, the user can always define `key` and `field_name` as per their wish. I have provided the option of not defining it and let it be automatically determined for the case of avoiding redundancy.
Use case, let's assume there are metrics which are similar in nature except for the `key`, `field_name` (like they can be of same type like resources metrics). In that case, it would be better to have a common configuration for them and let the `key` and `field_name` be automatically determined from the name (than creating 3-4 individual configurations).
I think that's the purpose of configuration to label the ones that are alike, right?
Also, `key` and `field_name` were optional field previously with `key` taking the value from name and `field_name` defaulting to `value` so I thought better not to change it.